src/lib/ThumbnailFactory.js

```diff
@@ -184,83 +208,77 @@ class ThumbnailFactory {
-     || canvas.imageResource;
+    if (resource.isCollection()) {
+      const firstManifest = resource.getManifests()[0];
+      if (firstManifest) return this.getSourceContentResource(firstManifest);
 
-    return (preferredCanvasResource && this.getResourceThumbnail(preferredCanvasResource))
-      || this.getThumbnail(canvas, { quirksMode: true, requireIiif: false });
-  }
+      return undefined;
+    }
 
-  /** */
-  getManifestThumbnail(manifest) {
-    const thumb = this.getThumbnail(manifest, { requireIiif: true });
-    if (thumb) return thumb;
+    if (resource.isManifest()) {
+      const miradorManifest = new MiradorManifest(resource);
+      const canvas = miradorManifest.startCanvas || miradorManifest.canvasAt(0);
+      if (canvas) return this.getSourceContentResource(canvas);
 
-    const miradorManifest = new MiradorManifest(manifest);
-    const canvas = miradorManifest.startCanvas || miradorManifest.canvasAt(0);
+      return undefined;
+    }
 
-    return (canvas && this.getIIIFThumbnail(canvas))
-      || this.getThumbnail(manifest, { quirksMode: true, requireIiif: false });
-  }
+    if (resource.isCanvas()) {
+      const image = ThumbnailFactory.getPreferredImage(resource);
+      if (image) return this.getSourceContentResource(image);
 
-  /** */
-  getCollectionThumbnail(collection) {
-    const thumb = this.getThumbnail(collection, { requireIiif: true });
-    if (thumb) return thumb;
+      return undefined;
+    }
 
-    const firstManifest = this.resource.getManifests()[0];
+    if (resource.getType() === 'image') {
+      return resource;
+    }
 
-    return (firstManifest && this.getManifestThumbnail(firstManifest))
-      || this.getThumbnail(collection, { quirksMode: true, requireIiif: false });
+    return undefined;
   }
 
-  /** */
+  /**
+   * Gets a thumbnail representing the resource.
+   * @return {Object|undefined} A thumbnail representing the resource, or undefined if none could
+   * be determined
+   */
   get() {
     if (!this.resource) return undefined;
 
-    if (this.resource.isCanvas()) return this.getIIIFThumbnail(this.resource);
-    if (this.resource.isManifest()) return this.getManifestThumbnail(this.resource);
-    if (this.resource.isCollection()) return this.getCollectionThumbnail(this.resource);
-    return this.getResourceThumbnail(this.resource, { requireIiif: true });
-  }
-}
+    // Determine which content resource we should use to derive a thumbnail
+    const sourceContentResource = this.getSourceContentResource(this.resource);
+    if (!sourceContentResource) return undefined;
 
-/** */
-function getBestThumbnail(resource, iiifOpts) {
-  return new ThumbnailFactory(resource, iiifOpts).get();
+    // Special treatment for external resources
+    if (typeof sourceContentResource === 'string') return { url: sourceContentResource };
+
+    return this.iiifThumbnailUrl(sourceContentResource);
+  }
 }
 
-export default getBestThumbnail;
+export default ThumbnailFactory;
```

Comment:
This now changes the export of this file which we would consider a breaking change only available in the next major version. Can we maintain the previous export API? I think this was an intentional consideration.

Reply:
Maybe @cbeer has some thoughts on the intention behind the API export here?